src/internal/connector/onedrive/collections.go

```diff
@@ -405,6 +461,25 @@ func (c *Collections) UpdateCollections(
 			return err
 		}
 
+		oneDrivePath, err := path.ToOneDrivePath(collectionPath)
+		if err != nil {
+			return clues.Wrap(err, "invalid path for backup")
+		}
+
+		if len(oneDrivePath.Folders) == 0 {
+			// path for root will never change
+			prevCollectionPath = collectionPath
+		} else {
+			prevCollectionPathStr, ok := oldPaths[collectionID]
+			if ok {
+				prevCollectionPath, err = path.FromDataLayerPath(prevCollectionPathStr, false)
+				if err != nil {
+					return clues.Wrap(err, fmt.Sprintf("invalid previous path for collection %s", prevCollectionPathStr))
+				}
+			}
+
+		}
```

Comment:
this whole block of code can be moved to where we create a new container when adding the item (roughly line 600). It's only needed in that case and having it so far away makes it hard to determine where these variables are actually used